### PR TITLE
modules/bootkube: fix etcd-operator startup with S3 backups and nobody

### PR DESCRIPTION
**Pull request by Quentin-M**
Fixes https://github.com/coreos/tectonic-installer/issues/959.
See issue for details.

### CODE REVIEW

**Comment by squat** [comment]
@Quentin-M Is this ported from upstream?

**Comment by Quentin-M** [comment]
@squat Thanks for checking! The [example](https://github.com/coreos/etcd-operator/blob/master/example/deployment.yaml) manifest upstream runs as root, so this does not apply.

**Comment by kans** [comment]
@diegs should this be backported?

**Comment by kans** [comment]
@Quentin-M any idea if this problem exists elsewhere too?

**Comment by diegs** [comment]
Sorry, I was unaware of this behavior and it appears to not have test coverage.

Looks like this should be backported, or added to known issues with the workaround (add this manifest change between `make plan` and `make apply`).

File: modules/bootkube/resources/experimental/manifests/etcd-operator.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ spec:
               valueFrom: 
                 fieldRef: 
                   fieldPath: metadata.name
+            - name: HOME
```
Comment (diegs) [inline modules/bootkube/resources/experimental/manifests/etcd-operator.yaml]: I don't love that the etcd-operator writes to $HOME/.aws, but if that's an unavoidable side-effect of using AWS then so be it.

**Comment by sstarcher** [comment]
looks like this issue is back in newer versions of etcd operator I'm running 0.5.2 and it's trying to use
```/var/tmp/etcd-operator/aws```
where aws is the credential.

```
time="2017-09-13T21:57:32Z" level=info msg="etcd-operator Version: 0.5.2"
time="2017-09-13T21:57:32Z" level=info msg="Git SHA: 0fa2f7d"
time="2017-09-13T21:57:32Z" level=info msg="Go Version: go1.8.1"
time="2017-09-13T21:57:32Z" level=info msg="Go OS/Arch: linux/amd64"
time="2017-09-13T21:57:32Z" level=info msg="Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"etcd-operator", UID:"8b2da31f-98ce-11e7-bbf2-0aa4c304ed3c", APIVersion:"v1", ResourceVersion:"1545", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' etcd-operator-525807187-xghfx became leader"
2017-09-13 21:57:52.494409 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
time="2017-09-13T21:57:52Z" level=error msg="cluster failed to setup: new S3 storage failed: mkdir /var/tmp/etcd-operator/aws: permission denied" cluster-name=kube-etcd pkg=cluster
time="2017-09-13T21:57:52Z" level=warning msg="fail to handle event: ignore failed cluster (kube-etcd). Please delete its CR" pkg=controller
error: unexpected EOF```

**Pull request merged**